### PR TITLE
Complete padding of `CausalLMBatch` when there exists batch bucketing 

### PR DESCRIPTION
**Pull request by kaixuanliu**
Add padding to batches of `top_n_tokens` and `requests` to `from_pb` function when there is batch bucketing

### CODE REVIEW

**Comment by kaixuanliu** [comment]
@yuanwu2017 @regisss pls help review. Original implementation will crash during warmup when running
`docker run -d -p 8080:80 --privileged -v /data/huggingface/hub:/data --runtime=habana -e HUGGING_FACE_HUB_TOKEN=hf_*** -e USE_FLASH_ATTENTION=true -e FLASH_ATTENTION_RECOMPUTE=true -e no_proxy=localhost -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none -e TEXT_GENERATION_SERVER_IGNORE_EOS_TOKEN=true -e ENABLE_HPU_GRAPH=true -e LIMIT_HPU_GRAPH=true -e PT_HPU_ENABLE_LAZY_COLLECTIVES=true -e MAX_BATCH_SIZE=1 --cap-add=sys_nice --ipc=host tgi_gaudi --model-id meta-llama/Meta-Llama-3.1-70B-Instruct --sharded true --num-shard 8 --max-input-tokens 5000 --max-total-tokens 32000 `

File: server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -548,7 +599,11 @@ def from_pb(
         position_ids = attention_mask.long().cumsum(-1) - 1
         position_ids.masked_fill_(attention_mask == 0, 1)
 
-
+        old_bs = len(requests)
+        last_request = requests[-1] if old_bs > 0 else None
+        for idx in range(old_bs, new_bs):
+            top_n_tokens.append(-1)
+            requests.append(last_request)
```
Comment (yuanwu2017) [inline server/text_generation_server/models/causal_lm.py]: requests should not append the dummy request.

**Review by yuanwu2017 — COMMENTED** [review]
requests should not append dummy request.

File: server/text_generation_server/models/causal_lm.py (hunk 1 above)
Comment (kaixuanliu) [reply server/text_generation_server/models/causal_lm.py]: Yes, you are right. Have updated the code.

**Review by yuanwu2017 — APPROVED** [review]

**Review by regisss — APPROVED** [review]
LGTM!
